### PR TITLE
Keycloak authentication

### PR DESCRIPTION
**Pull request by andpopov**
Part of: #24

### CODE REVIEW

**Comment by andpopov** [comment]
@olenagerasimova @dgarus please review

File: src/main/java/com/artipie/auth/AuthFromKeycloak.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    /**
+     * Ctor.
+     * @param config Configuration
+     */
+    public AuthFromKeycloak(final Configuration config) {
+        this.config = config;
+    }
+
+    @Override
+    @SuppressWarnings("PMD.OnlyOneReturn")
```
Comment (olenagerasimova) [inline src/main/java/com/artipie/auth/AuthFromKeycloak.java:39]: this suppression is not required, method has only one return....

File: src/test/java/com/artipie/auth/AuthFromKeycloakTest.java [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+/**
+ * Test for {@link AuthFromKeycloak}.
+ *
+ * @since 0.28
+ * @checkstyle IllegalCatchCheck (500 lines)
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ */
+@SuppressWarnings("PMD.AvoidCatchingThrowable")
+@Testcontainers
+public class AuthFromKeycloakTest {
```
Comment (olenagerasimova) [inline src/test/java/com/artipie/auth/AuthFromKeycloakTest.java:55]: Disable this test on windows, github win actions does not have docker

**Review by olenagerasimova — CHANGES_REQUESTED** [review]
@andpopov thanks, just minor comments and please, pay attention to the CI, it's failing now

**Review by olenagerasimova — DISMISSED** [review]

File: src/main/java/com/artipie/auth/AuthFromKeycloak.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        try {
+            final AuthorizationResponse response = client
+                .authorization(username, password, "openid")
+                .authorize(new AuthorizationRequest());
+            final AccessToken token = TokenVerifier.create(response.getToken(), AccessToken.class)
+                .getToken();
+            final Set<String> roles = new HashSet<>();
+            roles.addAll(AuthFromKeycloak.realmRoles(token));
+            roles.addAll(AuthFromKeycloak.clientRoles(token));
+            return Optional.of(new User(username, roles.stream().toList()));
```
Comment (dgarus) [inline src/main/java/com/artipie/auth/AuthFromKeycloak.java:50]: `new User(username, roles.stream().toList())` can be replaced by `new User(username, roles)`

**Review by dgarus — CHANGES_REQUESTED** [review]
Thanks.
There are minor comments, please have a look.

**Review by dgarus — APPROVED** [review]
lgtm

**Review by olenagerasimova — APPROVED** [review]

**Review by dgarus — APPROVED** [review]

**Pull request merged**